src/Bicep.Core/Syntax/VariableBlockSyntax.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Bicep.Core.Parsing;
+
+namespace Bicep.Core.Syntax
+{
+    public class VariableBlockSyntax : SyntaxBase
+    {
+        public VariableBlockSyntax(Token openParen, IEnumerable<SyntaxBase> children, SyntaxBase closeParen)
```

Comment:
This is similar to `ForVariableBlockSyntax` - which has been specifically designed to expect exactly 2 arguments. As a follow up to this PR, I think it would make sense to remove `ForVariableBlockSyntax` and use `VariableBlockSyntax` for the `(item, i)` in for loops - I'm just wary that this PR is already pretty large.